src/tls/openssl/tls.c

```diff
@@ -1078,30 +1090,7 @@ int tls_set_ciphers(struct tls *tls, const char *cipherv[], size_t count)
 
 
 /**
- * Set the server name on a TLS Connection, using TLS SNI extension.
- *
- * @param tc         TLS Connection
- * @param servername Server name
- *
- * @return 0 if success, otherwise errorcode
- */
-int tls_set_servername(struct tls_conn *tc, const char *servername)
-{
-	if (!tc || !servername)
-		return EINVAL;
-
-	if (1 != SSL_set_tlsext_host_name(tc->ssl, servername)) {
-		DEBUG_WARNING("tls: SSL_set_tlsext_host_name error\n");
-		ERR_clear_error();
-		return EPROTO;
-	}
-
-	return 0;
-}
```

Comment:
I am not sure if it is a good practice to remove public functions.

it is probably best to keep it...

Reply:
Yes, maybe we should keep it. But on the other side:
- tls_peer_set_verify_host()
- tls_set_servername()
- tls_set_verify_server()

is very confusing. Our TLS expert told me that SSL_set1_host() and SSL_set_tlsext_host_name() should be used together. Also setting TLS method to SSL_VERIFY_PEER. If we have only one function which does this for the TLS client makes the re API simpler.

Should we keep the functions anyway? If yes, I would add more details to the doxygen of these functions.